### ISSUE TITLE
ssl_verify_mode setting ignored

### ISSUE THREAD

**Issue by haidangwa** [hashicorp/terraform#7712]
### Terraform Version
Terraform v0.6.16

### Affected Resource(s)
- null_resource
- aws_instance
- _haven't tried other resources_

If this issue appears to affect multiple resources, it may be an issue with Terraform's core, so please mention this.

### Terraform Configuration Files
The AMI being provisioned has chef-client 12.11.18 pre-installed.

```hcl
  provisioner "chef" {
    node_name              = "taecs-${var.chef_organization}-be0-${var.chef_environment}"
    environment            = "${var.chef_environment}"
    run_list               = ["role[chefha_backend]"]
    server_url             = "${var.chef_server_url}"
    secret_key             = "${file("${var.chef_secret_key}")}"
    validation_key         = "${file("${var.chef_validator_key}")}"
    validation_client_name = "${var.chef_validator_client_name}"
    ssl_verify_mode        = "${var.ssl_verify_mode}"
    skip_install           = true
    log_to_file            = true
  }
```
ssl_verify_mode variable is set to `ssl_verify_mode = "verify_none"`

### Expected Behavior
The chef provisioner should run chef-client with ssl_verify_mode configured to `:verify_none` on the target chef-client node; thus, no ssl validation is carried out.

### Actual Behavior
chef log shows the following SSL certificate hostname mismatch:
```
[2016-07-20T02:48:08-04:00] ERROR: SSL Validation failure connecting to host: internal-us-west-elb-2069234877.us-west-2.elb.amazonaws.com - hostname "internal-us-west-elb-2069234877.us-west-2.elb.amazonaws.com" does not match the server certificate
```

### Steps to Reproduce
1. configure an aws_instance with the chef provisioner, as described above. Be sure to set the `ssl_verify_mode` to `verify_none`
2. `terraform apply`

### Important Factoids
I've tried to specify the value of ssl_verify_mode, as follows: `ssl_verify_mode = "verify_none"` and `ssl_verify_mode = ":verify_none"`. Both get the same error.

**Comment by svanharmelen** [comment]
@haidangwa can you post the `client.rb` that is created on the target node?

**Comment by haidangwa** [comment]
This is with my terraform variables configured as `ssl_verify_mode="verify_none"`:

**Comment by haidangwa** [comment]
Oddly, setting this in my terraform vars to `ssl_verify_mode=":verify_none"` worked this time. My testing must have had some PEBCAK problems before. So, this might just boil down to poor documentation here, https://www.terraform.io/docs/provisioners/chef.html.

**Comment by haidangwa** [comment]
Oddly, when I went to create a new instance to show you the client.rb when I specify `ssl_verify_mode = ":verify_none"` (with the value as a colon inside the String value), it seemed to work this time. So, this might boil down to poor documentation here, https://www.terraform.io/docs/provisioners/chef.html.

**Comment by haidangwa** [comment]
Oddly, when I went to create a new instance to show you the client.rb when I specify ssl_verify_mode = ":verify_none" (with the value as a colon inside the String value), it seemed to work this time. So, this might boil down to poor documentation here, https://www.terraform.io/docs/provisioners/chef.html.

Below is the client.rb with `ssl_verify_mode = ":verify_none"`:

**Comment by svanharmelen** [comment]
@haidangwa check... Thanks for the feedback and let me wipe up a small PR to improve the UX on this one... I'll refer to this issue is the PR.

**Comment by svanharmelen** [comment]
Should be fixed (improved) by #7769
